week-1-ts-ecosystem/w01e04-light-me-up/src/task1/LightBulb.ts

```diff
@@ -0,0 +1,15 @@
+import { PowerSource } from "./PowerSource";
+
+export class LightBulb {
+  name: string;
+  protected readonly powerConsumption = 20;
+
+  constructor(name: string) {
+    this.name = name;
+  }
+
+  plugIn(powerSource: PowerSource) {
```

Comment:
Najczęściej powerSource dajesz przez `constructor` i wtedy korzystasz z pola `this.powerSource` - jako referencji do niego.

Tutaj akurat jedynie robimy `.consume` ale nabiera to większego znaczenia jak w jeszcze innych metodach z niego chcesz korzystać.